app/js/actions/addDrugOrder.js

```diff
@@ -6,23 +6,24 @@ import {
 import axiosInstance from '../config';
 import networkError from './networkError';
 import loading from './loading';
+import activeOrderAction from './activeOrderAction';
 
-export const postDrugOrderLoading = status => ({
-  type: POST_DRUG_ORDER_LOADING,
-  status,
-});
-export const postDrugOrderSuccess = () => ({
+export const postDrugOrderSuccess = response => ({
   type: POST_DRUG_ORDER_SUCCESS,
+  response,
 });
+
 export const postDrugOrderFailure = error => ({
   type: POST_DRUG_ORDER_FAILURE,
   error,
 });
-export const postDrugOrder = (ordersPayload, patientUuid, careSetting) => (dispatch) => {
+
+export const postDrugOrder = (ordersPayload, careSetting, patientUuid) => (dispatch) => {
   dispatch(loading('POST_DRUG_ORDER', true));
-  return axiosInstance.get(`encounter`, ordersPayload)
-    .then(() => {
-      dispatch(postDrugOrderSuccess());
+  return axiosInstance.post(`encounter`, ordersPayload)
+    .then((response) => {
+      dispatch(postDrugOrderSuccess(response));
+      dispatch(activeOrderAction(careSetting, patientUuid));
```

Comment:
Oh I see you're dispatching active order action, but rather than make multiple calls to the API, wouldn't it be better to handle this in the store (reducer)?

Reply:
@zeze111 Handling it from the store was not working, that is why I did a dispatch from the action.

Reply:
Okay, sounds good